File: problems/3639-minimum-time-to-activate-string/solution.py
class Solution:
    def minTime(self, s: str, order: List[int], k: int) -> int:
        n = len(s)

        def count_valid_substrings(temp_s):
            count = 0
            for i in range(n):
                for j in range(i, n):
                    substring = temp_s[i:j+1]
                    if '*' in substring:
                        count += 1
            return count

        for t in range(n):
            temp_s = list(s)
            for i in range(t + 1):
                temp_s[order[i]] = '*'
            temp_s = "".join(temp_s)
            
            if count_valid_substrings(temp_s) >= k:
                return t
        
        return -1

File: problems/3639-minimum-time-to-activate-string/solution.cpp
class Solution {
public:
    int minTime(string s, vector<int>& order, int k) {
        int n = s.length();
        for (int t = 0; t < n; ++t) {
            string temp = s;
            for (int i = 0; i <= t; ++i) {
                temp[order[i]] = '*';
            }
            long long count = 0;
            for (int i = 0; i < n; ++i) {
                for (int j = i; j < n; ++j) {
                    bool valid = false;
                    for (int l = i; l <= j; ++l) {
                        if (temp[l] == '*') {
                            valid = true;
                            break;
                        }
                    }
                    if (valid) {
                        count++;
                    }
                }
            }
            if (count >= k) {
                return t;
            }
        }
        return -1;
    }
};